src/public/services/__tests__/FieldVerificationService.test.ts

```
@@ -0,0 +1,136 @@
+      const expected: FetchNewTransactionResponse = {
+        expireAt: new Date().toJSON() as JsonDate,
+        transactionId: 'some transaction id',
+      }
+      const mockFormId = 'mockFormId'
+
+      // Act
+      const actualPromise = createTransactionForForm(mockFormId)
+      mockAxios.mockResponse({ data: expected })
+      const actual = await actualPromise
+
+      // Assert
+      expect(actual).toEqual(expected)
+      expect(mockAxios.post).toHaveBeenCalledWith(TRANSACTION_ENDPOINT, {
+        formId: mockFormId,
+      })
+    })
+    it('should successfully return empty transaction data when returned data is empty', async () => {
+      // Arrange
+      const expected: FetchNewTransactionResponse = {}
+      const mockFormId = 'mockFormId2'
+
+      // Act
+      const actualPromise = createTransactionForForm(mockFormId)
+      mockAxios.mockResponse({ data: expected })
+      const actual = await actualPromise
+
+      // Assert
+      expect(actual).toEqual(expected)
+      expect(mockAxios.post).toHaveBeenCalledWith(TRANSACTION_ENDPOINT, {
+        formId: mockFormId,
+      })
+    })
+  })
+
+  describe('triggerSendOtp', () => {
+    it('should call endpoint successfully', async () => {
+      // Arrange
+      const mockTransactionId = 'mockTransactionId'
+      const mockFieldId = 'someFieldId'
+      const mockAnswer = 'blablabla'
+
+      // Act
+      const actualPromise = triggerSendOtp({
+        transactionId: mockTransactionId,
+        answer: mockAnswer,
+        fieldId: mockFieldId,
+      })
+      mockAxios.mockResponse({ status: 201, data: 'Created' })
+      await actualPromise
+
+      // Assert
+      expect(mockAxios.post).toHaveBeenCalledWith(
+        `${TRANSACTION_ENDPOINT}/${mockTransactionId}/otp`,
+        {
+          fieldId: mockFieldId,
+          answer: mockAnswer,
+        },
+      )
+    })
+  })
+
+  describe('verifyOtp', () => {
+    it('should return verified signature on success', async () => {
```

do we want to test for error cases?

---

The errors are not being handled in the service (yet) though, that's why I didn't include that.